lua/Network.lua

```diff
@@ -0,0 +1,651 @@
+
+		vectorDistance[cell] = distance
+	end)
+
+	return vectorDistance
+end
+
+local function checksDistancePointTarget(lines, arrayTargetLine)
+	local countTarget = 0
+	local point = {}
+	local pointTarget = {}
+	local keyPoint = true
+
+	while arrayTargetLine[countTarget] do
+        local nPoints = 0
+		pointTarget[arrayTargetLine[countTarget]] = {}
+
+		forEachCell(lines, function(line)
+			local bePoint = getBELine(line)
+			local result
+
+
+			local Point1 = tl:castGeomToSubtype(bePoint.p1)
+			local Point2 = tl:castGeomToSubtype(bePoint.p2)
+
+			local finalDistanceA = Point1:distance(Point2)
+			local finalDistanceB = 0
+
+			result = checkDistance(line, finalDistanceB, arrayTargetLine[countTarget])
+
+			if keyPoint then 
+				table.insert(point, Point1)
+				nPoints = #point
+			else
+				nPoints = nPoints + 1 
+			end
+
+			pointTarget[arrayTargetLine[countTarget]][point[nPoints]] = result
+
+			result = checkDistance(line, finalDistanceA, arrayTargetLine[countTarget])
+
+			if keyPoint then 
+				table.insert(point, Point2)
+				nPoints = #point
+			else
+				nPoints = nPoints + 1 
+			end
+
+			pointTarget[arrayTargetLine[countTarget]][point[nPoints]] = result
+
+		end)
+
+		countTarget = countTarget + 1
+		keyPoint = false
+
+	end
+
+	local nw = {
+		point = point,
+		target = arrayTargetLine,
+		distance = pointTarget
+	}
+
+	return nw
+end
+
+local function checksInterconnectedNetwork(data)
+	local ncellRed = 0
+	local warning = false
+
+	forEachCell(data.lines, function(cellRed)
+		local geometryR = tl:castGeomToSubtype(cellRed.geom:getGeometryN(0))
+		local bePointR = getBELine(cellRed)
+		local lineValidates = false
+		local differance = math.huge
+		local distance
+		local redPoint
+		local nConect = 0
+		cellRed.route = {}
+
+		for j = 0, 1 do
+			if j == 0 then
+				redPoint = tl:castGeomToSubtype(bePointR.p1)
+			else
+				redPoint = tl:castGeomToSubtype(bePointR.p2)
+			end
+
+			local ncellBlue = 0
+
+			forEachCell(data.lines, function(cellBlue)
+				local geometryB = tl:castGeomToSubtype(cellBlue.geom:getGeometryN(0))
+
+				if geometryR:crosses(geometryB) then
+					customWarning("The lines '"..geometryB:toString().."' and '"..geometryR:toString().."' crosses")
+					warning = true
+				end
+
+				local bePointB = getBELine(cellBlue)
+				local bluePoint
+
+				for i = 0, 1 do
+
+					if i == 1 then
+
+						bluePoint = tl:castGeomToSubtype(bePointB.p1)
+					else
+						bluePoint = tl:castGeomToSubtype(bePointB.p2)
+					end
+
+					if ncellRed == ncellBlue then break end 
+
+					distance = redPoint:distance(bluePoint)
+
+					if distance <= data.error then
+						table.insert(cellRed.route, cellBlue)
+						lineValidates = true
+						nConect = nConect + 1
+					end
+
+					if differance > distance then
+						differance = distance
+					end
+				end
+
+				ncellBlue = ncellBlue + 1
+			end)
+		end
+
+		if not lineValidates then
+			customError("line do not touch, They have a differance of: "..differance)
```

Comment:
`("Lines do not touch each other. They have a distance of "..difference..".")`
You should show the line id that does not touch the rest of the network.